utils.go

```diff
@@ -52,26 +60,26 @@ func FindFilterFunction(funcName string) (string, error) {
 		return "", err
 	}
 
+	availableFilterFunctions.Lock()
+	defer availableFilterFunctions.Unlock()
+
 	// Cache available filter functions if necessary
-	if len(availableFilterFunctions) == 0 {
+	if len(availableFilterFunctions.cache) == 0 {
 		funcs, err := tracefs.ReadFile("available_filter_functions")
 		if err != nil {
 			return "", err
 		}
-		availableFilterFunctions = strings.Split(string(funcs), "\n")
-		for i, name := range availableFilterFunctions {
-			splittedName := strings.Split(name, " ")
-			name = splittedName[0]
-			splittedName = strings.Split(name, "\t")
-			name = splittedName[0]
-			availableFilterFunctions[i] = name
+		availableFilterFunctions.cache = strings.Split(string(funcs), "\n")
```

Comment:
I *think* this approach might still keep the entire file in memory, since the used strings functions don't create copies, but just slices pointing at the same backing array.

If you use a `bufio.Scanner`, I think that would fix this issue.

Reply:
Or you could use `copy` at the end to actually create a new copy of the just the names.

Reply:
Can you write a benchmark to showcase your improvements?